packages/asset-swapper/contracts/src/WooPPSampler.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.6;
+pragma experimental ABIEncoderV2;
+import "./interfaces/IWooPP.sol";
+import "./SamplerUtils.sol";
+import "./ApproximateBuys.sol";
+
+
+contract WooPPSampler is SamplerUtils, ApproximateBuys{
+
+   function query(
+        uint amountIn, 
+        address tokenIn, 
+        address tokenOut,
+        address pool
+    ) internal view returns (uint256 amountOut) {
+        if (amountIn == 0) { 
+            return 0; 
+        }
+        address quoteToken = IWooPP(pool).quoteToken();
+        if (tokenIn == quoteToken) {
+            amountOut = IWooPP(pool).querySellQuote(tokenOut, amountIn);
+        } else if (tokenOut == quoteToken) {
+            amountOut = IWooPP(pool).querySellBase(tokenIn, amountIn);
+        } else {
+            uint quoteAmount = IWooPP(pool).querySellBase(tokenIn, amountIn);
```

Comment:
isn't this backward? 

`querySellQuote` first then `querySellBase` ?

Reply:
@kyu-c  don't we need to go from:
base -> quote ->base  == `querySellBase` -> `querySellQuote`

Reply:
(discussed separately)

`quoteToken` is a stable token. (i thought it was the other way around)